src/HeaderAddPlugin.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Adds headers to the request only if the request does not already own it.
```

Comment:
Hm, I see the problem. I would still keep this functionality here. I could imagine a HeaderDefaults plugin which only adds stuff if necessary. WDYT?